### PR TITLE
Clarify & update "Debug WebView2 apps with Visual Studio"

### PR DESCRIPTION
**Pull request by mikehoffms**
Rendered article for review:
* **Debug WebView2 apps with Visual Studio**
   * https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/how-to/debug-visual-studio?branch=pr-en-us-3233
   * [Public preview](https://github.com/MicrosoftDocs/edge-developer/blob/user/mikehoffms/png-wv2-debug-vs/microsoft-edge/webview2/how-to/debug-visual-studio.md)
   * [Before/Live](https://learn.microsoft.com/microsoft-edge/webview2/how-to/debug-visual-studio)

Changes made by this PR:
* Fixed Issue https://github.com/MicrosoftDocs/edge-developer/issues/2775 so that the last png shows a VS screenshot, not a VS Code screenshot.
* Added "Summary of high-level steps" at top.
* Updated the article to focus on VS 2022, not 2019.
* Added all specific steps to reproduce the screenshots; provided the specific steps to follow to use the main wv2 sample (while also allowing to apply to user's own app code base).
   * Instead of abstract steps referring to user's own code base, w/ pngs showing an unspecified code base (turns out to be main wv2 sample).
   * Made article maintainable & testable.
* Added the missing prerequisite steps (as links) for setting up a new machine that are needed trying to follow the article: 
   * Install VS.
      * Which version? was unclear; do install VS 2022 (just mention that 2019 is supported).
   * Clone samples repo.
      * In order to provide & follow specific steps.  Enables maintenance & testing the article.
   * Install C++ workload.
      * Or other platform workload.
      * May or may not have been done by user previous to arriving at this article.
   * Compile & run main sample.
      * To use the main sample here, it's best if you already got it compiled & running per sample's article.  In this article, just build on that milestone.
      * Starting state: Have the **WebView2Samples** solution open, **WebView2APISample** sample app (project) compiles, is not running.
* In repo, changed 60-char divider comments to standard 40-char (above h4s & some tab h5s).

AB#52806681 - PR Tracking item
AB#45948351 - Issue Tracking item
AB#54124902 - Future test/fix article to hit breakpoint.

### CODE REVIEW

File: microsoft-edge/webview2/how-to/debug-visual-studio.md [hunk 1]
```diff
@@ -35,59 +35,96 @@ To debug your code, first install and set up the script debugger tools in Visual
...
+   ![The 'Debugging' configuration property in Visual Studio](./debug-visual-studio-images/enb-js.png)<!-- WebView2APISample -->
 
 
 <!-- ====================================================================== -->
 ## Debug your WebView2 app
 
 After doing the above setup, debug your WebView2 app, as follows.
 
-1. To set a breakpoint in your source code, hover to the left of the line number, and click to set a breakpoint.  The JS/TS debug adapter doesn't perform source path mapping.  You must open the exact same path associated with your WebView2.
+1. To set a breakpoint in your source code, hover to the left of the line number, and click to set a breakpoint.  For example, open `WebView2Samples\SampleApps\WebView2APISample\assets\ScenarioJavaScriptDebugIndex.js`, and set a breakpoint on line 2: `console.log("onAddClick+");`, in the body of `function onAddClick() {`:
```
Comment (mikehoffms) [inline microsoft-edge/webview2/how-to/debug-visual-studio.md:107]: not clear that setting this breakpoint does anything, in these steps
@victorhuangwq

File: microsoft-edge/webview2/how-to/debug-visual-studio.md [hunk 2]
```diff
@@ -22,11 +22,11 @@ When you debug your app in Visual Studio with the native debugger attached, pres
 <!-- ====================================================================== -->
 ## Requirements
 
-*  To debug scripts, the app must be launched from within Visual Studio.
+*  To debug scripts, the app must be launched from within Visual Studio.  You can install both Visual Studio 2019 (to work with most of the WebView2 sample apps) and Visual Studio 2022 (for example, for your own WebView2 app).
```
[the remark this answers is not in the document]
Comment (victorhuangwq) [reply microsoft-edge/webview2/how-to/debug-visual-studio.md:25]: I chatted with @oggy22 and @peiche-jessica, all sample apps should work with 2022, and also supports 2019.

I think for the rest of the document, we should assume that:
- Sample Apps work on 2022
- Development should be done on 2022

We can mention that 2019 is supported

File: microsoft-edge/webview2/how-to/debug-visual-studio.md [hunk 3]
```diff
@@ -22,11 +22,11 @@ When you debug your app in Visual Studio with the native debugger attached, pres
...
 
 *  You cannot attach a debugger to a running WebView2 process.
 
-*  Install Visual Studio 2019 version 16.4 Preview 2 or later.
+*  Install Visual Studio 2019 (version 16.4 Preview 2 or later).  If that's not done yet, see [If you're installing Visual Studio now](#if-youre-installing-visual-studio-now), below.
```
[the remark this answers is not in the document]
Comment (victorhuangwq) [reply microsoft-edge/webview2/how-to/debug-visual-studio.md:29]: At the same time, we should ask them to install 2022

File: microsoft-edge/webview2/how-to/debug-visual-studio.md [hunk 4]
```diff
@@ -35,59 +35,96 @@ To debug your code, first install and set up the script debugger tools in Visual
...
 ## Install the JavaScript diagnostics component
 
-First, install the **JavaScript diagnostics** component in **Desktop development with C++**, as follows.
+First, install the **JavaScript diagnostics** component in the **Desktop development with C++** workload of Visual Studio, as follows.
+
+
+<!-- ------------------------------ -->
+#### If you're installing Visual Studio now
+
+1. To work with the WebView2 samples, download Visual Studio 2019 (version 16.4 Preview 2 or later), from [Downloads](https://visualstudio.microsoft.com/downloads/), and then run the downloaded installer file.  To work with your own WebView2 app, you might want to do these steps for Visual Studio 2022 as well.
```
Comment (victorhuangwq) [inline microsoft-edge/webview2/how-to/debug-visual-studio.md:44]: We should just ask them to use Visual Studio 2022. Don't need to ask them to have both versions

File: microsoft-edge/webview2/how-to/debug-visual-studio.md [hunk 5]
```diff
@@ -35,59 +35,96 @@ To debug your code, first install and set up the script debugger tools in Visual
...
+See also:
+* [Install Visual Studio](./machine-setup.md#install-visual-studio) in _Set up your Dev environment for WebView2_.
+
+Continue with [Enable script debugging for WebView2 apps](#enable-script-debugging-for-webview2-apps), below.
+
+
+<!-- ------------------------------ -->
+#### If you've already installed Visual Studio
+
+To work with most of the WebView2 samples, use Visual Studio 2019.  To work with your own WebView2 app, you might want to do these steps for Visual Studio 2022 as well.
```
Comment (victorhuangwq) [inline microsoft-edge/webview2/how-to/debug-visual-studio.md:61]: Same situation - Visual Studio 2022 will suffice for both sample apps and own WebView2 app

File: microsoft-edge/webview2/how-to/debug-visual-studio.md [hunk 6]
```diff
@@ -35,59 +35,96 @@ To debug your code, first install and set up the script debugger tools in Visual
...
    ![Visual Studio: Modifying values in the 'Individual components' tab](./debug-visual-studio-images/indiv-comp.png)
 
+1. Click the **Modify** button.
+
 
 <!-- ====================================================================== -->
 ## Enable script debugging for WebView2 apps
 
-Second, enable script debugging for WebView2 apps, as follows.
+Second, enable script debugging for WebView2 apps, as follows.  You can open your own WebView2 app in Visual Studio 2019 or 2022.  The following steps walk you through:
```
Comment (victorhuangwq) [inline microsoft-edge/webview2/how-to/debug-visual-studio.md:87]: open app in 2022

File: microsoft-edge/webview2/how-to/debug-visual-studio.md (hunk 1 above)
Comment (victorhuangwq) [reply microsoft-edge/webview2/how-to/debug-visual-studio.md:107]: The breakpoint only triggers when this section of the code is executed. I believe that this should be triggered upon clicking the "Add a new item" button in step 3. 

But I don't think there's a step that asks the user to click on it, and thus this breakpoints functionality is demonstrated.

File: microsoft-edge/webview2/samples/webview2apissample.md [hunk 7]
```diff
@@ -72,12 +72,30 @@ To use Visual Studio 2017, you might also need to install a recent Windows SDK o
 <!-- ====================================================================== -->
 ## Step 4 - Open the solution in Visual Studio
 
-1. On your local drive, open the `.sln` file in Visual Studio:
+1. Open Visual Studio 2022.  (Or, if you have Visual Studio 2019 installed, you can use that version.)  For example, in Windows, in the Start menu, search for **Visual Studio 2022** app, and then click **Open**:
```
Comment (victorhuangwq) [inline microsoft-edge/webview2/samples/webview2apissample.md:75]: I think this step is pretty verbose. We can just state
"Open Visual Studio 2022", and drop the screenshot

File: microsoft-edge/webview2/samples/webview2apissample.md [hunk 8]
```diff
@@ -72,12 +72,30 @@ To use Visual Studio 2017, you might also need to install a recent Windows SDK o
...
 
+   ![The "Create a new branch" dialog](./webview2apissample-images/create-a-new-branch.png)
+
+1. Click the **Solution Explorer** tab, in the lower right, and then in **Solution Explorer**, double-click **WebView2Samples.sln**.
+
+   Or, the following approach requires more navigation: Select **File** > **Open** > **Project/Solution** > `WebView2Samples.sln`, such as (on Windows): `C:\Users\myUsername\GitHub\WebView2Samples\SampleApps\WebView2Samples.sln`
+
+   Example paths where you may have cloned the WebView2Samples repo to:
+   *  `<your-repos-directory>/WebView2Samples/SampleApps/WebView2Samples.sln`
    *  `<your-repos-directory>/WebView2Samples-main/SampleApps/WebView2Samples.sln`
```
Comment (victorhuangwq) [inline microsoft-edge/webview2/samples/webview2apissample.md:99]: Actually, I think the following steps are not quite correct. I agree that the original description was a little bit confusing and could be clarified. 

I think the general steps can be summarized be as follows:
1. Open Visual Studio 2022
2. Select `Open a project or solution`
3. Navigate to where you cloned the WebView2Samples repository, and select the `WebView2Samples.sln`. E.g. `<some-directory>\WebView2Samples\SampleApps\WebView2\WebView2Samples.sln`

File: microsoft-edge/webview2/samples/webview2apissample.md [hunk 9]
```diff
@@ -140,14 +160,41 @@ Solution Explorer shows several projects, including the **WebView2APISample** pr
...
+1. In the **Solution Configurations** dropdown list, select **Debug** (or, **Release**).
 
 1. In the **Solution Platforms** dropdown list, select **x86**, **x64**, or **ARM64**.
 
-1. In **Solution Explorer**, right-click the **WebView2APISample** project, and then select **Build**.
+1. In **Solution Explorer**, right-click the **WebView2APISample (Visual Studio 2019)** project, and then select **Build**.
 
    ![The WebView2APISample project selected in Solution Explorer](./webview2apissample-images/webview2apisample-project-selected.png)
 
+   The project is bold, indicating that it's the default in case you use keyboard shortcuts to build.
```
Comment (victorhuangwq) [inline microsoft-edge/webview2/samples/webview2apissample.md:171]: @oggy22 @peiche-jessica Do users need to retarget using 2019 build tools?

**Review by victorhuangwq — CHANGES_REQUESTED** [review]

File: microsoft-edge/webview2/samples/webview2apissample.md (hunk 9 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/samples/webview2apissample.md]: reverted changes in PR other than debug article, will re-apply/ followup in other PR

File: microsoft-edge/webview2/samples/webview2apissample.md (hunk 8 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/samples/webview2apissample.md]: reverted changes in PR other than debug article, will re-apply/ followup in other PR

File: microsoft-edge/webview2/samples/webview2apissample.md (hunk 7 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/samples/webview2apissample.md]: reverted changes in PR other than debug article, will re-apply/ followup in other PR

File: microsoft-edge/webview2/how-to/debug-visual-studio.md (hunk 6 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/how-to/debug-visual-studio.md:97]: updated a few lines in section

File: microsoft-edge/webview2/how-to/debug-visual-studio.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mikehoffms) [reply microsoft-edge/webview2/how-to/debug-visual-studio.md:39]: resequenced, expanded, reworded

File: microsoft-edge/webview2/how-to/debug-visual-studio.md [hunk 10]
```diff
@@ -6,98 +6,181 @@ ms.author: msedgedevrel
...
+   onAddClick+
+   onAddClick-
+   End
+   ```
+
+1. In the webpage in the app, click the **Add a new item** popup button.
+
+   The breakpoint in the `onAddClick` function is hit, and the app is paused in the debugger.
+
+1. Switch to the Visual Studio window, and examine the debugger panes.
```
Comment (mikehoffms) [inline microsoft-edge/webview2/how-to/debug-visual-studio.md:155]: When walking through the end-to-end steps in revised article, I never did get the debugger to halt at breakpoint, after I clicked Add Item in the running wv2 sample app.  
@captainbrosset 
 
Nice to do: Patrick do the steps in PR's article, see if hits breakpoint.  See if article gives the usage info that the article ideally should give, about why we're setting a breakpoint, and what exactly eg we can do in the debugger panes (I never arrived at useful debugger panes UX).

File: microsoft-edge/webview2/how-to/debug-visual-studio.md [hunk 11]
```diff
@@ -6,98 +6,181 @@ ms.author: msedgedevrel
...
+Continue with [Enable script debugging for WebView2 apps](#enable-script-debugging-for-webview2-apps), below.
+
 
 <!-- ====================================================================== -->
 ## Enable script debugging for WebView2 apps
 
-Second, enable script debugging for WebView2 apps, as follows.
+Second, enable script debugging for WebView2 apps, as follows.  The following steps walk you through:
+* Opening the Win32APISample sample in Visual Studio 2022.
+* Working within your own working branch of the WebView2Samples repo, where some of the project files are modified during updating to Visual Studio 2022.
```
Comment (mikehoffms) [inline microsoft-edge/webview2/how-to/debug-visual-studio.md:91]: where did the "create branch" steps go?  check commits, & check PR about main sample.

"The following steps walk you through ... Working within your own working branch of the WebView2Samples repo, where some of the project files are modified during updating to Visual Studio 2022."

Comment (mikehoffms) [reply microsoft-edge/webview2/how-to/debug-visual-studio.md]: removed mention of branch; leveraged following main-sample article, stated expected setup state

File: microsoft-edge/webview2/how-to/debug-visual-studio.md (hunk 10 above)
Comment (captainbrosset) [reply microsoft-edge/webview2/how-to/debug-visual-studio.md:155]: I did the steps from the tutorial, but couldn't get the breakpoint to hit either. It looks like we're maybe missing some configuration?
Here's what I'm seeing when setting a breakpoint:

image

Comment (mikehoffms) [reply microsoft-edge/webview2/how-to/debug-visual-studio.md:155]: i still suspect, check the settings in
https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/how-to/debug-visual-studio-images/enb-js.png
a lot going on there, that could affect UX.
@captainbrosset 
seems like need real, actual explanation of what's going on: we're installing another debugger, and trying to set this debugger to be used, for some scenario.  that background explanation is proved to be needed & missing here.

Comment (captainbrosset) [reply microsoft-edge/webview2/how-to/debug-visual-studio.md:155]: Lets use AB#54124902 to track this.

**Review by captainbrosset — APPROVED** [review]

**Pull request merged**